### DISCUSSION TITLE
unable to pull an image with default podman machine on macOS

### DISCUSSION THREAD

**Discussion by benoitf** [containers/podman#16657] [General]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Pulling an image reports the error:
```
Error: writing blob: adding layer with blob "sha256:96840549f90fa24b9df1ab5335fb527b2b1fb7825afd0f48a4d33e3d3e154a05": processing tar file(potentially insufficient UIDs or GIDs available in user namespace (requested 0:1000640000 for /usr/local/bin/tkn): Check /etc/subuid and /etc/subgid if configured locally and run podman-system-migrate: lchown /usr/local/bin/tkn: invalid argument): exit status 1
```

**Steps to reproduce the issue:**

1. start a fresh podman machine (in my case it's a `4.3.0` podman on the remote machine and 4.3.1 on the client side)

2. run the command: `podman pull quay.io/devfile/universal-developer-image:ubi8-latest`

**Describe the results you received:**
```
Error: writing blob: adding layer with blob "sha256:96840549f90fa24b9df1ab5335fb527b2b1fb7825afd0f48a4d33e3d3e154a05": processing tar file(potentially insufficient UIDs or GIDs available in user namespace (requested 0:1000640000 for /usr/local/bin/tkn): Check /etc/subuid and /etc/subgid if configured locally and run podman-system-migrate: lchown /usr/local/bin/tkn: invalid argument): exit status 1
```

**Describe the results you expected:**

image being pulled.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
(paste your output here)
```

**Output of `podman info`:**

```
(paste your output here)
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
(paste your output here)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by benoitf** [comment]
probably it's due rootless (uids > 65536) as it's working in rootful mode, but the error message could be:

Podman is using rootless mode, blablabla, needs to use rootful mode to circumvent the issue

**Comment by Luap99** [comment]
I would not recommend root. As the error says you can add more uids to /etc/sub{u,g}id and run podman system migrate.

To be honest since the machine is only used by one user I see no reason why we shouldn't just use a much bigger range when we init the machine.

**Comment by ibuziuk** [comment]
looks like there is no `podman system migrate` in 4.3.1

```
ibuziuk@ibuziuk-mac ~ % podman system --help 
Manage podman

Description:
  Manage podman

Usage:
  podman system [command]

Available Commands:
  connection  Manage remote API service destinations
  df          Show podman disk usage
  info        Display podman system information
  prune       Remove unused data

```

**Comment by benoitf** [comment]
@ibuziuk you will need to execute on the machine first. `podman machine ssh` then you can execute the `system` commands. podman on the host is the "remote one" and has not the system commands
